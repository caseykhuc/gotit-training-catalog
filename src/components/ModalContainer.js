import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import modalKeys from '../constants/modelKeys';
import RegisterModal from './Modal/RegisterModal';
import SignInModal from './Modal/SignInModal';

export const ModalContainer = ({ modal }) => {
  const renderModal = () => {
    switch (modal.current) {
      case modalKeys.REGISTER_MODAL:
        return <RegisterModal />;
      case modalKeys.SIGNIN_MODAL:
        return <SignInModal />;
      default:
        return '';
    }
  };

  // return <div>ModalContainer {modal.current}</div>;
  return (
    <div style={{
      position: 'absolute',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
    }}
    >
      {renderModal()}
    </div>
  );
};

ModalContainer.propTypes = {
  modal: PropTypes.shape({
    current: PropTypes.string,
  }).isRequired,
}

const mapStateToProps = (state) => ({
  modal: state.modal,
});

export default connect(mapStateToProps)(ModalContainer);
