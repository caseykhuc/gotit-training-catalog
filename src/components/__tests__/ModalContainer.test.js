import React from 'react';
import { shallow } from 'enzyme';
import { ModalContainer } from '../ModalContainer';
import modalKeys from '../../constants/modelKeys';

describe('component/ModalContainer', () => {
  let props;
  let wrapper;
  let authBtn;

  const update = () => {
    wrapper.update();
  };
  const setup = () => {
    wrapper = shallow(<ModalContainer {...props} />);
    update();
  };
  beforeEach(() => {
    props = {
      modal: {
        current: 'SAMPLE_MODAL',
      },
    }
  });
  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();

    props.modal.current = modalKeys.DELETE_MODAL;
    setup();
    expect(wrapper).toMatchSnapshot();

    props.modal.current = modalKeys.REGISTER_MODAL;
    setup();
    expect(wrapper).toMatchSnapshot();

    props.modal.current = modalKeys.SIGNIN_MODAL;
    setup();
    expect(wrapper).toMatchSnapshot();
  });
})
