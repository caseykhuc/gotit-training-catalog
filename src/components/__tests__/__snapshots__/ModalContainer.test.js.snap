// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/ModalContainer should render correctly 1`] = `ShallowWrapper {}`;

exports[`component/ModalContainer should render correctly 2`] = `ShallowWrapper {}`;

exports[`component/ModalContainer should render correctly 3`] = `ShallowWrapper {}`;

exports[`component/ModalContainer should render correctly 4`] = `ShallowWrapper {}`;

exports[`component/ModalContainer should render correctly 5`] = `ShallowWrapper {}`;

exports[`component/ModalContainer should render correctly 6`] = `ShallowWrapper {}`;

exports[`component/ModalContainer should render correctly 7`] = `ShallowWrapper {}`;
